File: manual/code/test_suites/python/python/test_shapely_buffer.py
import unittest
from function_shapely_buffer import GeometryBuffer

class TestGeometryBuffer(unittest.TestCase):
    def setUp(self):
        self.geometry_buffer = GeometryBuffer()

    def test_point_buffer(self):
        geom = "POINT(0 0)"
        distance = 1.0
        expected_start = "POLYGON"
        result = self.geometry_buffer.calculate_buffer(geom, distance)
        self.assertTrue(result.startswith(expected_start))

    def test_polygon_buffer(self):
        geom = "POLYGON((0 0, 2 0, 1 2, 0 0))"
        distance = 0.5
        expected_start = "POLYGON"
        result = self.geometry_buffer.calculate_buffer(geom, distance)
        self.assertTrue(result.startswith(expected_start))

    def test_negative_buffer(self):
        geom = "POLYGON((0 0, 4 0, 2 4, 0 0))"
        distance = -0.5
        expected_start = "POLYGON"
        result = self.geometry_buffer.calculate_buffer(geom, distance)
        self.assertTrue(result.startswith(expected_start))

    def test_empty_geometry(self):
        geom = "GEOMETRYCOLLECTION EMPTY"
        distance = 1.0
        expected = "GEOMETRYCOLLECTION EMPTY"
        self.assertEqual(self.geometry_buffer.calculate_buffer(geom, distance), expected)

    def test_invalid_geometry(self):
        geom = "INVALID GEOMETRY"
        distance = 1.0
        expected = "Error: Could not process the input geometry."
        self.assertEqual(self.geometry_buffer.calculate_buffer(geom, distance), expected)

if __name__ == "__main__":
    unittest.main()

File: data/java/tests/test_shapely_buffer.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_shapely_buffer {
    private function_shapely_buffer geometryBuffer;

    @BeforeEach
    void setUp() {
        geometryBuffer = new function_shapely_buffer();
    }

    @Test
    void test_point_buffer() {
        String geom = "POINT(0 0)";
        float distance = 1.0f;
        String expectedStart = "POLYGON";
        String result = geometryBuffer.calculate_buffer(geom, distance);
        assertTrue(result.startsWith(expectedStart));
    }

    @Test
    void test_polygon_buffer() {
        String geom = "POLYGON((0 0, 2 0, 1 2, 0 0))";
        float distance = 0.5f;
        String expectedStart = "POLYGON";
        String result = geometryBuffer.calculate_buffer(geom, distance);
        assertTrue(result.startsWith(expectedStart));
    }

    @Test
    void test_negative_buffer() {
        String geom = "POLYGON((0 0, 4 0, 2 4, 0 0))";
        float distance = -0.5f;
        String expectedStart = "POLYGON";
        String result = geometryBuffer.calculate_buffer(geom, distance);
        assertTrue(result.startsWith(expectedStart));
    }

    @Test
    void test_empty_geometry() {
        String geom = "GEOMETRYCOLLECTION EMPTY";
        float distance = 1.0f;
        String expected = "GEOMETRYCOLLECTION EMPTY";
        assertEquals(expected, geometryBuffer.calculate_buffer(geom, distance));
    }

    @Test
    void test_invalid_geometry() {
        String geom = "INVALID GEOMETRY";
        float distance = 1.0f;
        String expected = "Error: Could not process the input geometry.";
        assertEquals(expected, geometryBuffer.calculate_buffer(geom, distance));
    }
}